smartsim/_core/launcher/step/dragonStep.py

```diff
@@ -90,3 +100,157 @@ def _get_exe_args_list(run_setting: DragonRunSettings) -> t.List[str]:
         exe_args = run_setting.exe_args
         args: t.List[str] = exe_args if isinstance(exe_args, list) else [exe_args]
         return args
+
+
+class DragonBatchStep(Step):
+    def __init__(
+        self,
+        name: str,
+        cwd: str,
+        batch_settings: t.Union[SbatchSettings, QsubBatchSettings],
+    ) -> None:
+        """Initialize a Slurm Sbatch step
+
+        :param name: name of the entity to launch
+        :type name: str
+        :param cwd: path to launch dir
+        :type cwd: str
+        :param batch_settings: batch settings for entity
+        :type batch_settings: SbatchSettings
+        """
+        super().__init__(name, cwd, batch_settings)
+        self.steps: t.List[Step] = []
+        self.managed = True
+        self.batch_settings = batch_settings
+        self._request_file_name = "requests.json"
+
+    def get_launch_cmd(self) -> t.List[str]:
+        """Get the launch command for the batch
+
+        :return: launch command for the batch
+        :rtype: list[str]
+        """
+        if isinstance(self.batch_settings, SbatchSettings):
+            script = self._write_sbatch_script()
+            return [self.batch_settings.batch_cmd, "--parsable", script]
+        if isinstance(self.batch_settings, QsubBatchSettings):
+            script = self._write_qsub_script()
+            return [self.batch_settings.batch_cmd, script]
+
+        raise SSUnsupportedError(
+            "DragonBatchStep only support SbatchSettings and QsubBatchSettings"
+        )
+
+    def add_to_batch(self, step: Step) -> None:
+        """Add a job step to this batch
+
+        :param step: a job step instance e.g. DragonStep
+        :type step: Step
+        """
+        self.steps.append(step)
+        logger.debug(f"Added step command to batch for {step.name}")
+
+    @staticmethod
+    def _dragon_entrypoint_cmd(request_file: str) -> str:
+        """Return command needed to run the Dragon entrypoint"""
+        cmd = [
+            f"{sys.executable}",
+            "-m",
+            "smartsim._core.entrypoints.dragon_client",
+            "+submit",
+            f"{request_file}",
+            "\n",
```

Comment:
What does this newline do? If it's meant to alter the joined output, move there to be clear that it's not part of command?

```
foo = " ".join(cmd) 
return f"{foo}\n"
```

Reply:
It seems odd that the command would require a newline at all. Must dig...